emu_sv/hamiltonian.py

```diff
@@ -132,18 +133,19 @@ def _create_diagonal(self) -> torch.Tensor:
             diagonal elements.
         """
         diag = torch.zeros(
-            (2,) * self.nqubits, dtype=torch.complex128, device=self.omegas.device
+            2**self.nqubits, dtype=torch.complex128, device=self.deltas.device
         )
 
         for i in range(self.nqubits):
-            i_fixed = diag.select(i, 1)
-            i_fixed -= self.deltas[i]  # add the delta term for this qubit
+            diag = diag.reshape((2**i, 2, -1))  # -1 means "remaining"
+            i_fixed = diag.select(dim=1, index=1)
```

Comment:
I think it is more explicit to use normal slicing syntax now that the number of dimensions is known:
`i_fixed = diag[:, 1, :]`

Reply:
Nice, done

Reply:
somewhere I saw a discussion about performance difference between select and slicing.